main.js

```diff
@@ -0,0 +1,104 @@
+var postsDiv = document.getElementsByClassName('posts')[0];
+
+document.getElementById('submit-post').addEventListener('click', function () {
+    postsDiv.append(createPost(document.getElementById('post-name'), document.getElementById('post-text')));
+});
+
+var createPost = function(name, text) {
+    var postName = name.value;
+    var postText = text.value;
+
+    var newPost = document.createElement('div');
+    newPost.setAttribute('class', 'post');
+
+    var commentDiv = document.createElement('div');
+    commentDiv.setAttribute('class', 'comments hidden');
+    var comments = document.createElement('div');
+    commentDiv.appendChild(comments);
+
+    var postTextParagraph = document.createElement('p');
+    var postTextElement = document.createTextNode(postText + ' - Posted by: ' + postName);
+    postTextParagraph.appendChild(postTextElement);
+    
+    var showCommentsButton = document.createElement('button');
+    showCommentsButton.setAttribute('type', 'button');
+    showCommentsButton.setAttribute('class', 'btn-xs btn-info show-comments');
+    showCommentsButton.innerHTML = 'show comments';
+
+    showCommentsButton.addEventListener('click', function () {
+        console.log(showCommentsButton.parentElement.querySelector('.comments'));
+        if(showCommentsButton.parentElement.querySelector('.comments').classList.contains('hidden')) {
+            showCommentsButton.parentElement.querySelector('.comments').className = 'comments';

```

Comment:
could be rewritten to be simpler using ternary
showCommentsButton.parentElement.querySelector('.comments').className += showCommentsButton.parentElement.querySelector('.comments').classList.contains('hidden') ? 'comments' : ' hidden'